### PR TITLE
Convert to ES6 that is supported on Node 4, commonjs modules and remove Babel

### PR DESCRIPTION
**Pull request by corbinu**
This removes Babel, converts to commonjs modules and removes other ES6 features not supported on Node 4. I think it was all destructuring.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -45,17 +45,18 @@
...
-import Referencer from './referencer';
-import Reference from './reference';
-import Variable from './variable';
-import Scope from './scope';
-import { version } from '../package.json';
+const ScopeManager = require('./scope-manager').default;
+const Referencer = require('./referencer').default;
+const Reference = require('./reference').default;
+const Variable = require('./variable').default;
+const Scope = require('./scope').default;
```
Comment (not-an-aardvark) [inline src/index.js]: Can you use imports like `require('./foo')` here instead of `require('./foo').default`? Then the modules could use `module.exports = Foo` instead of `module.exports.default = Foo`.

File: src/scope.js [hunk 2]
```diff
@@ -21,14 +21,15 @@
   (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
   THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */
+"use strict";
 
-import { Syntax } from 'estraverse';
-import Map from 'es6-map';
+const Syntax = require('estraverse').Syntax;
+const Map = require('es6-map');
```
Comment (not-an-aardvark) [inline src/scope.js]: I think this import is unnecessary now since `Map` is included in Node 4. (That can be changed in a separate PR if necessary though.)

File: src/scope.js [hunk 3]
```diff
@@ -632,16 +633,31 @@ export class FunctionScope extends Scope {
...
 
-export class ClassScope extends Scope {
+class ClassScope extends Scope {
     constructor(scopeManager, upperScope, block) {
         super(scopeManager, 'class', upperScope, block, false);
     }
 }
 
+module.exports = {
+    "default": Scope,
```
Comment (not-an-aardvark) [inline src/scope.js]: Since this isn't using modules anymore, it might be better to use `module.exports.Scope` instead of `module.exports.default`.

File: src/scope.js (hunk 2 above)
Comment (corbinu) [inline src/scope.js]: @not-an-aardvark That is planned as a separate PR

File: src/index.js (hunk 1 above)
Comment (corbinu) [inline src/index.js]: I was planning to do a style clean up PR at the end that would include this. Wanted to keep changes in each PR to a minimum but can change it now if people would prefer

**Comment by soda0289** [comment]
Looks good. Nothing too drastic which is nice. I would remove the check node.body commit until the other pull request is approved.

File: src/index.js (hunk 1 above)
Comment (not-an-aardvark) [inline src/index.js]: I don't really have a strong opinion either way, but I think it makes more sense to change this here since it's a direct result of migrating from ES6 modules.

Comment (corbinu) [inline src/index.js]: ok will do!
